tls/s2n_connection.h

```diff
@@ -85,6 +101,11 @@ struct s2n_connection {
     uint8_t actual_protocol_version;
     uint8_t actual_protocol_version_established;
 
+    /* Certificate Authentication and Verification Parameters */
+    s2n_cert_auth_type client_cert_auth_type;
```

Comment:
I'm against renaming `client_cert_auth_type` to something that does not contain `client` or `client_cert` in it. This variable determines which states in the state machine are taken, and only affects the Client Certificate States.

If the flag is set to `NONE`, the normal handshake is executed, if it is set to `REQUIRED` then the `CLIENT_CERT_REQUEST`, `CLIENT_CERT`, and `CLIENT_CERT_VERIFY` states are executed on **_both the client and the server_**. Renaming it `peer_auth_type` would be incorrect since this flag does not affect how the Client authenticates the Server.